vars/piperPipelineStageIntegration.groovy

```diff
@@ -29,14 +34,25 @@ void call(Map parameters = [:]) {
         .mixinGeneralConfig(script.commonPipelineEnvironment, GENERAL_CONFIG_KEYS)
         .mixinStageConfig(script.commonPipelineEnvironment, stageName, STEP_CONFIG_KEYS)
         .mixin(parameters, PARAMETER_KEYS)
+        .addIfEmpty('npmExecuteScripts', script.commonPipelineEnvironment.configuration.runStep?.get(stageName)?.npmExecuteScripts)
         .use()
 
     piperStageWrapper (script: script, stageName: stageName) {
 
         // telemetry reporting
         utils.pushToSWA([step: STEP_NAME], config)
 
-        echo "${STEP_NAME}: No default stage implementation is provided for this stage. You can extend the stage using the provided stage extension mechanism."
-
+        boolean publishResults = false
+        try {
+            if (config.npmExecuteScripts) {
+                publishResults = true
+                npmExecuteScripts script: script
```

Comment:
Shouldn't the `runScripts` parameter be set to a certain value, e.g., `ci-it-frontend`, when calling `npmExecuteScripts`? Otherwise all scripts defined for `runScripts` in the config.yml or possibly no script will be executed if none are defined or?

Reply:
This can be configured by the user in the stage config or we (cloud sdk pipeline) provide a default.